Directory.Build.targets

```diff
@@ -0,0 +1,37 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+
+  <Target
+    Name="EnsureJavaScriptCodeUpToDate"
+    BeforeTargets="BeforeBuild"
+    Inputs="$(MSBuildThisFileDirectory)yarn.lock"
+    Outputs="$(RootIntDir)\JsUpToDateCheck\yarn.lock"
+    >
+
+    <Message Text="Checking if Yarn.lock file is 'up to date' by running a 'dry-run' version of yarn install and checking the exit code" />
+    <Exec 
+      ContinueOnError="True"
+      Command="npx yarn install --offline --cache-folder $(RootIntDir)\JsUpToDateCheck\Cache"
```

Comment:
>$(RootIntDir)\JsUpToDateCheck\Cache [](start = 57, length = 35)

It can be just current folder "."

Reply:
No, this is a targets file and the CWD is not guarnateed in msbuild. By default the CWD is the folder of where the .csproj of .vcxproj lives. This is not correct here since we need to use the root.  Any in-proc task in msbuild can change CWD and affect anything that runs after it. I chose a specific fixed (non project-specific) folder so that if a yarn does actually ends up writing to that folder for some config reason we are not polluting the source tree, but only the `built` folder.